src/trials.jl

```diff
@@ -339,168 +355,194 @@ Base.show(io::IO, t::TrialRatio) = _show(io, t)
+    mintime = minimum(t.times)
+    printstyled(io, prettytime(mintime); color=:default, bold=true)
+    print(io, ", ")
+    printstyled(io, "median "; color=medcolor)
+    medtime = median(t.times)
+    printstyled(io, prettytime(medtime); color=medcolor, bold=true)
+    # printstyled(io, " (½)"; color=medcolor)
+    print(io, ", ")
+    printstyled(io, "mean "; color=avgcolor)
+    avgtime = mean(t.times)
+    printstyled(io, prettytime(avgtime); color=avgcolor, bold=true)
+    # printstyled(io, " (*)"; color=avgcolor)
+    print(io, ", ")
+    print(io, showpercentile, "ᵗʰ ")
+    quantime = quantile(t.times, showpercentile/100)
+    printstyled(io, prettytime(quantime); bold=true)
+    println(io)
+
+    printstyled(io, pad, "│", boxspace; color=boxcolor)
+    println(io, allocsstr)
+
+    if !all(iszero, t.gctimes)
+        # Mean GC time is just that; then we take the percentage of the mean time
+        printstyled(io, pad, "│", boxspace; color=boxcolor)
+        _avgc = mean(t.gctimes)
+        print(io, "GC time: mean ", prettytime(_avgc))
+        printstyled(io, " (", prettypercent(_avgc / avgtime), ")"; color=avgcolor)
+
+        # Maximum GC time is _not_ taken as the GC time of the slowst run, maximum(t).
+        # The percentage shown is of the same max-GC run, again not the percentage of longest time.
+        # Of course, very often the slowest run is due to GC, and these distinctions won't matter.
+        _t, _i = findmax(t.gctimes)
+        println(io, ", max ", prettytime(_t), " (", prettypercent(_t / t.times[_i]), ")")
+    end
 
     # Histogram
 
-    histquantile = 0.99
-    # The height and width of the printed histogram in characters.
-    histheight = 2
-    histwidth = 42 + lmaxtimewidth + rmaxtimewidth
-
-    histtimes = times[1:round(Int, histquantile*end)]
-    histmin = get(io, :histmin, first(histtimes))
-    histmax = get(io, :histmax, last(histtimes))
-    logbins = get(io, :logbins, nothing)
-    bins = bindata(histtimes, histwidth - 1, histmin, histmax)
-    append!(bins, [1, floor((1-histquantile) * length(times))])
-    # if median size of (bins with >10% average data/bin) is less than 5% of max bin size, log the bin sizes
-    if logbins === true || (logbins === nothing && median(filter(b -> b > 0.1 * length(times) / histwidth, bins)) / maximum(bins) < 0.05)
-        bins, logbins = log.(1 .+ bins), true
-    else
-        logbins = false
+    logbins = get(io, :logbins, nothing) === true
+    caption = logbins ? ("log(counts) from " * samplesstr) : samplesstr
+
+    # The height and width of the printed histogram in characters:
+    histheight = 2    
+    _nonhistwidth = 5 + length(boxspace)
+    _minhistwidth = 18 + length(caption)
+    histwidth = max(_minhistwidth, min(90, displaysize(io)[2]) - _nonhistwidth)
+    # This should fit it within your terminal, but stops growing at 90 columns. Below about
+    # 55 columns it will stop shrinking, by which point the first line has already wrapped.
```

Comment:
This auto-sizes the histogram to fit your window, but stops at 90 characters.

I think it works OK, but is it a good idea? That results in quite a big histogram, maybe it should stay smaller? Or maybe it should fill arbitrarily wide windows?